src/gui/editors/PianoRoll.cpp

```diff
@@ -4313,20 +4289,39 @@ void PianoRoll::enterValue( NoteVector* nv )
 	}
 }
 
+void PianoRoll::updateXScroll()
+{
+	m_leftRightScroll->setGeometry(
+		m_whiteKeyWidth,
+		height() - SCROLLBAR_SIZE,
+		width() - m_whiteKeyWidth,
```

Comment:
I'll try to add that to this PR, since it looks like it shouldn't be too difficult. I can also adjust the Song Editor and Automation Editor scrollbars while I'm at it. The Automation Editor updates its horizontal scrollbar's range in the paint event just like Piano Roll did before I improved it, so I should probably apply the same improvement to that class too.